third_party/move/move-compiler-v2/src/pipeline/split_critical_edges_processor.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This pass splits critical edges with empty blocks.
+//! A critical edge is an edge where the source node has multiple successors,
+//! and the target node has multiple predecessors.
+//!
+//! Side effects: clear existing annotations.
+//!
+//! Prerequisites: no call instructions have abort actions.
+//!
+//! Postconditions: no critical edges in the control flow graph.
+
+use move_model::{ast::TempIndex, model::FunctionEnv};
+use move_stackless_bytecode::{
+    function_target::FunctionData,
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{AttrId, Bytecode, Label},
+};
+use std::collections::{BTreeMap, BTreeSet};
+
+pub struct SplitCriticalEdgesProcessor {}
+
+impl FunctionTargetProcessor for SplitCriticalEdgesProcessor {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        fun_env: &FunctionEnv,
+        mut data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if cfg!(debug_assertions) {
+            Self::check_precondition(&data);
+        }
+        if fun_env.is_native() {
+            return data;
+        }
+        let mut transformer = SplitCriticalEdgesTransformation::new(std::mem::take(&mut data.code));
+        transformer.transform();
+        data.code = transformer.code;
+        data.annotations.clear();
+        if cfg!(debug_assertions) {
+            Self::check_postcondition(&data.code);
+        }
+        data
+    }
+
+    fn name(&self) -> String {
+        "SplitCriticalEdgesProcessor".to_owned()
+    }
+}
+
+impl SplitCriticalEdgesProcessor {
+    /// Checks the precondition of the transformaiton; cf. module documentation.
+    fn check_precondition(data: &FunctionData) {
+        for instr in &data.code {
+            if matches!(instr, Bytecode::Call(_, _, _, _, Some(_))) {
+                panic!("precondition violated: found call instruction with abort action")
+            }
+        }
+    }
+
+    /// Checks the postcondition of the transformation; cf. module documentation.
+    fn check_postcondition(code: &[Bytecode]) {
+        use move_stackless_bytecode::stackless_control_flow_graph::{
```

Comment:
Why this local use? If its a type it should go to the top of the file, cargo fmt will take care to optimize them. The use case where you have local `use` is if you want to do something like `use Bytecode::*`. Reason is that if someone looks at the dependency section of the file, they want to see them nicely summarized.

Reply:
Done. (There used to be a `#[cfg(debug_assertion)]` attribute on this method, so I used the local use to avoid unnecessary use when debug_assertion attribute is not set.